experimental/instrumentation-session-recording/src/instrumentation.ts

```diff
@@ -0,0 +1,101 @@
+import type { eventWithTime } from '@rrweb/types';
+import { record, type recordOptions } from 'rrweb';
+
+import { BaseInstrumentation, VERSION } from '@grafana/faro-core';
+
+import { defaultSessionRecordingInstrumentationOptions } from './const';
+import type { SessionRecordingInstrumentationOptions } from './types';
+
+const faroSessionRecordingEventName = 'faro.session_recording.event';
+
+export class SessionRecordingInstrumentation extends BaseInstrumentation {
+  readonly name = '@grafana/faro-web-session-recording';
+  readonly version = VERSION;
+
+  private stopFn: { (): void } | null = () => {};
+  private isRecording: boolean = false;
+  private options: SessionRecordingInstrumentationOptions = defaultSessionRecordingInstrumentationOptions;
+
+  constructor(options: SessionRecordingInstrumentationOptions = {}) {
+    super();
+
+    this.options = {
+      ...defaultSessionRecordingInstrumentationOptions,
+      ...options,
+    };
+  }
+
+  initialize(): void {
+    if (this.isRecording) {
+      this.logWarn('Session recording is already running');
+      return;
+    }
+
+    this.startRecording();
```

Comment:
It looks like the instrumentation automatically starts recording when it initializes. This means that if included as a regular instrumentation in the Faro configuration, every session will be recorded. Is this what we want, or should there be some sampling?

Reply:
That's correct; I wanted to keep sampling out of scope for now. I'd love to add it in subsequent PRs.